### ISSUE TITLE
Product Gallery Vimeo Videos Don't Play Over  HTTPS

### ISSUE THREAD

**Issue by pioneerathletics** [magento/magento2#8574]
### Preconditions

1. Magento version 2.1.4 CE (PHP- 7.0.13)
2. Base URL set to https

### Steps to reproduce

1. Changed the Base URL to https in order to make the entire site secure
2. Added a product with a Vimeo Video in the product gallery (used https url here as well)

### Expected result

1. Product video should play as it did when base URL was set to http

### Actual result

1. The video would not play
2.  According to Dev Tools Console output -- Blocked due to Mixed Content error "( This request has been blocked; the content must be served over HTTPS.)"

**Comment by orlangur** [comment]
You should see in Dev Tools Console which content was served via HTTP. There would be no error if everything was served via HTTPS in fact.

**Comment by pioneerathletics** [comment]
The content served via http is the video player. Even though, the url to Vimeo is https. That's why it is strange that the video is being blocked. 

```
Mixed Content: The page at 'https://website.com/example' was loaded over HTTPS, but requested an insecure resource 'http://player.vimeo.com/video/1222222222222?api=1&player_id=vimeo12222222222222222&autoplay=1'. This request has been blocked; the content must be served over HTTPS.
(anonymous) @ jquery.js:5707
```

**Comment by smokygeek17** [comment]
same problem. any fix?

**Comment by pioneerathletics** [comment]
I fixed the issue by creating Magento_ProductVideo>web>js>load-player.js in my theme.
Then find line 320 in load-player.js and change http://player.vimeo.com/video/ to https://player.vimeo.com/video/

Now video plays as expected.

**Comment by korostii** [comment]
So, even though you've found a working workaround, it's still a bug in Magento CE? 

Wouldn't you want to have it fixed in Magento itself, as well? 
Just a friendly FYI: keeping the issue open might help with that, and a pull request is also an option.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-72278

**Comment by magento-engcom-team** [comment]
Hi @pioneerathletics the issue has been fixed in 2.1.10 and will be available soon
